examples/style.yaml

```diff
@@ -0,0 +1,68 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+# Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
+on:
+  push:
+    paths: ["R/**", "tests/**", "vignettes/**"]
+  pull_request:
+    paths: ["R/**", "tests/**", "vignettes/**"]
```

Comment:
These paths cover the majority of files but may exclude some given a non-traditional package structure. Would it be better to go by file type instead? For example:
`paths: ["**.r", "**.R", "**.rmd", "**.Rmd", "**.rmarkdown", "**.Rmarkdown", "**.Rnw", "**.rnw"]`

Reply:
That seems more in line with what we actually want, true. Plus `".Rprofile"`. Also, I don't know exactly what your use case is but since you seem to already use pre-commit, you can also just run it in the cloud with https://pre-commit.ci (it's not free though for private repos I think) or GitHub Actions to enforce the style, as described in https://github.com/lorenzwalthert/precommit/blob/main/NEWS.md#precommit-v020. That would save us from trouble like

* excluding files.
* identical config for pre-commit hook and CI run, e.g. for `scope`, file filter etc.
* identical environment (package versions etc.), since hooks with pre-commit.com use {renv} to specify hook dependencies.
* no additional config file to maintain.
* planned support for platforms other than GitHub to become more platform agnostic.
* ... 

Maybe we should update the docs of {precommit} to point out better that the CI integration can also be used without a local installation.

Anyways, for the cache, I just realised it's better to not use the hash of the commit as a key, because I don't think the cache mechanism will be able to pick up the latest SHA and use that cache. Instead, it will probably just fall back to some random cache key, or alphabetical order. Hence, I think it's better to make the cache date dependent, with fallbacks to yyyy-mm-dd, yyyy-mm.

Reply:
Thanks for bringing up running pre-commit on the cloud. I had looked at that briefly but then decided to just run locally. I will certainly look into it! Happy to make doc suggestions as well if I get confused—pre-commit has been a super useful tool!

> That seems more in line with what we actually want, true. Plus `".Rprofile"`

No harm in including as well but I can't think of too many workflows where someone would push `.Rprofile`. I think the only time I have pushed is when using `{renv}` (which is then a single line of code)...

> Anyways, for the cache, I just realised it's better to not use the hash of the commit as a key, because I don't think the cache mechanism will be able to pick up the latest SHA and use that cache. Instead, it will probably just fall back to some random cache key, or alphabetical order. Hence, I think it's better to make the cache date dependent, with fallbacks to yyyy-mm-dd, yyyy-mm.

Did a bit more digging into this. Github says [here](https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key) that:
> If there are multiple partial matches for a restore key, the action returns the most recently created cache.

So leveraging the SHA should be fine because it can track the most recently created SHA.

Reply:
> So leveraging the SHA should be fine because it can track the most recently created SHA. 

Thanks, if I was just capable of reading the docs  😉

Reply:
Hahaha no worries 😃